### PR TITLE
Router for extracting out application logic from witness creation

### PR DESCRIPTION
**Pull request by vaibhavchellani**

### CODE REVIEW

File: router/transfer.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	// TOOD remove and update the transfer
+	return core.LoadedBazooka.RollupUtils.CompressManyAirdropFromEncoded(&opts, data)
+}
+func (transfer transferTransaction) applyTxWithProof(accountMP core.AccountMerkleProof, tx core.Tx) (updatedAccount []byte, updatedRoot core.ByteArray, err error) {
+	opts := bind.CallOpts{From: config.OperatorAddress}
+	accMP, err := accountMP.ToABIVersion()
+	if err != nil {
+		return nil, core.ByteArray{}, err
+	}
+	updatedAccountBytes, updatedRoot, err := core.LoadedBazooka.RollupCaller.ApplyAirdropTx(&opts, accMP, tx.Data)
```
Comment (ChihChengLiang) [inline router/transfer.go:98]: Why we have `ApplyAirdropTx` in `transfer.go`?

File: router/transfer.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+func (transfer *transferTransaction) applyTxWithoutProof(tx core.Tx, accountID uint64) (updatedAccount []byte, updatedRoot core.ByteArray, err error) {
+	opts := bind.CallOpts{From: config.OperatorAddress}
+	var accountMP core.AccountMerkleProof
+	core.DBInstance.FetchMPWithID(accountID, &accountMP)
+	accMP, err := accountMP.ToABIVersion()
+	if err != nil {
+		return nil, core.ByteArray{}, err
+	}
+	updatedAccountBytes, updatedRoot, err := core.LoadedBazooka.RollupCaller.ApplyAirdropTx(&opts, accMP, tx.Data)
```
Comment (ChihChengLiang) [inline router/transfer.go:113]: Same question here

File: router/transfer.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+func (transfer *transferTransaction) compressTxs(txs []core.Tx) ([]byte, error) {
+	opts := bind.CallOpts{From: config.OperatorAddress}
+	var data [][]byte
+	for _, tx := range txs {
+		data = append(data, tx.Data)
+	}
+	// TOOD remove and update the transfer
+	return core.LoadedBazooka.RollupUtils.CompressManyAirdropFromEncoded(&opts, data)
```
Comment (ChihChengLiang) [inline router/transfer.go:90]: Why we use `CompressManyAirdropFromEncoded` in `transfer.go`?

**Review by ChihChengLiang — COMMENTED** [review]
Great work! This looks beautiful.

**Comment by vaibhavchellani** [comment]
Thanks @ChihChengLiang, this PR is not open for review yet, have made wireframes and started working on abstractions, that's why the issues with using airdrop in the transfer. Will fix those before the merge.

File: router/transfer.go (hunk 3 above)
Comment (vaibhavchellani) [reply router/transfer.go:90]: Will fix in a later commit, trying to connect things and see if the abstraction works.

**Comment by vaibhavchellani** [comment]
Will probably postpone this for later.

**Comment by vaibhavchellani** [comment]
Not relavent anymore

**Pull request closed without merging**